internal/controller/progressive/progressive.go

```diff
@@ -299,11 +299,16 @@ func processUpgradingChild(
 	// Otherwise, assess the previous child status.
 	assessment := childStatus.AssessmentResult
 	failureReason := childStatus.FailureReason
+	failureChildStatus := childStatus.ChildStatus.Raw
 	if childStatus.CanAssess() {
 		assessment, failureReason, err = controller.AssessUpgradingChild(ctx, rolloutObject, existingUpgradingChildDef)
 		if err != nil {
 			return false, 0, err
 		}
+		failureChildStatus, err = json.Marshal(existingUpgradingChildDef.Object["status"])
```

Comment:
nit: do you mind calling it `childStatus` instead of `failureChildStatus`? I'm thinking at this line of code we don't really know if it's a failure, success, or unknown.

Reply:
`childStatus` variable is already used there, so probably I can use `childSts`